pkg/tfgen/generate_go.go

```diff
@@ -438,6 +438,11 @@ func (g *goGenerator) emitResourceType(mod *module, res *resourceType) error {
 	if res.doc != "" {
 		g.emitDocComment(w, res.doc, res.docURL, "")
 	}
+	if !res.IsProvider() {
```

Comment:
```suggestion
	if !res.IsProvider() && res.info.DeprecationMessage != "" {
```

Reply:
(same throughout)

Reply:
Fixed up!

Reply:
Looks like this didn't actually get changed?

Reply:
all changed!

```
	if !res.IsProvider() && res.info.DeprecationMessage != "" {
		w.Writefmtln("warnings.warn(\"%s\", DeprecationWarning)", res.info.DeprecationMessage)
	}
~``